### ISSUE TITLE
Temperature Still Reverting to Celsius on 1.5.1

### ISSUE THREAD

**Issue by noltondyer** [hillaliy/homebridge-midea-air#70]
While controlling the device in HomeKit the first time, my AC sets to Fahrenheit, every time after, though, it defaults to Celsius. Plug-in debug mode is still reflecting a “false” value for useFahrenheit, even though “Default to Fahrenheit” is checked, and it’s set correctly in the config

**Comment by hillaliy** [comment]
I have OSK102 and I make all the tests on my AC.
At my AC everything works perfect.
The OSK103 is a little bit different.
The Default Fahrenheit help when you want that Fahrenheit will be still True when you restart Homebridge.

**Comment by hillaliy** [comment]
try 1.5.2-beta.1
Tell me if it fix the LED displey

**Comment by cdkii** [comment]
Updating to 1.5.2-beta.1 fixes the LED being turned off when controlling via the Home app, but the LED display is still being switched to Celsius.

**Comment by noltondyer** [comment]
Same here. Still defaulting to Celsius. What’s odd, the command being sent when displaying debug mode specifies “setFahrenheit: false” - even though the config correctly states true.

**Comment by cdkii** [comment]
Is there any info from the debug log would be useful in helping see what’s going on? I just tried looking and it all looks a bit insane to me with the 3 Midea U units and Cube dehumidifier info all scrolling past every 30 seconds.

Here’s part of it for one of the units... (The configuration is using “useFahrenheit": true)


```
[04/06/2022, 12:16:16] [midea-air] [updateValues] Send update command to: Bedroom AC (43980465157819)
[04/06/2022, 12:16:16] [midea-air] Triggered GET Active
[04/06/2022, 12:16:16] [midea-air] Triggered GET CurrentHumidifierDehumidifierState
[04/06/2022, 12:16:16] [midea-air] Triggered GET TargetHumidifierDehumidifierState
[04/06/2022, 12:16:16] [midea-air] Triggered GET CurrentRelativeHumidity
[04/06/2022, 12:16:16] [midea-air] Triggered GET RelativeHumidityDehumidifierThreshold
[04/06/2022, 12:16:16] [midea-air] Triggered GET WindSpeed
[04/06/2022, 12:16:16] [midea-air] Triggered GET SwingMode
[04/06/2022, 12:16:16] [midea-air] Triggered GET WaterLevel
[04/06/2022, 12:16:16] [midea-air] Triggered GET Active
[04/06/2022, 12:16:16] [midea-air] Triggered GET Current HeaterCooler State
[04/06/2022, 12:16:16] [midea-air] Triggered GET Target HeaterCooler State
[04/06/2022, 12:16:16] [midea-air] Triggered GET CurrentTemperature
[04/06/2022, 12:16:16] [midea-air] Triggered GET ThresholdTemperature
[04/06/2022, 12:16:16] [midea-air] Triggered GET ThresholdTemperature
[04/06/2022, 12:16:16] [midea-air] Triggered GET RotationSpeed
[04/06/2022, 12:16:16] [midea-air] Triggered GET SwingMode
[04/06/2022, 12:16:16] [midea-air] Triggered GET Temperature Display Units
[04/06/2022, 12:16:16] [midea-air] Triggered GET Active
[04/06/2022, 12:16:16] [midea-air] Triggered GET Current HeaterCooler State
[04/06/2022, 12:16:16] [midea-air] Triggered GET Target HeaterCooler State
[04/06/2022, 12:16:16] [midea-air] Triggered GET CurrentTemperature
[04/06/2022, 12:16:16] [midea-air] Triggered GET ThresholdTemperature
[04/06/2022, 12:16:16] [midea-air] Triggered GET ThresholdTemperature
[04/06/2022, 12:16:16] [midea-air] Triggered GET RotationSpeed
[04/06/2022, 12:16:16] [midea-air] Triggered GET SwingMode
[04/06/2022, 12:16:16] [midea-air] Triggered GET Temperature Display Units
[04/06/2022, 12:16:16] [midea-air] Triggered GET Active
[04/06/2022, 12:16:16] [midea-air] Triggered GET Current HeaterCooler State
[04/06/2022, 12:16:16] [midea-air] Triggered GET Target HeaterCooler State
[04/06/2022, 12:16:16] [midea-air] Triggered GET CurrentTemperature
[04/06/2022, 12:16:16] [midea-air] Triggered GET ThresholdTemperature
[04/06/2022, 12:16:16] [midea-air] Triggered GET ThresholdTemperature
[04/06/2022, 12:16:16] [midea-air] Triggered GET RotationSpeed
[04/06/2022, 12:16:16] [midea-air] Triggered GET SwingMode
[04/06/2022, 12:16:16] [midea-air] Triggered GET Temperature Display Units
[04/06/2022, 12:16:21] [midea-air] Triggered SET Active To: 0
[04/06/2022, 12:16:21] [midea-air] [sendUpdateToDevice] Header + Command: [90,90,1,16,90,0,32,0,1,0,0,0,189,179,57,14,12,5,20,20,29,129,0,0,0,16,0,0,0,4,2,0,0,1,0,0,0,0,0,0,170,30,172,0,0,0,0,0,3,2,64,0,40,102,127,127,0,60,0,0,0,0,0,0,0,0,0,0,0,108,189,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
[04/06/2022, 12:16:21] [midea-air] Triggered GET Active
[04/06/2022, 12:16:21] [midea-air] Triggered GET Active
[04/06/2022, 12:16:21] [midea-air] Triggered GET CurrentHumidifierDehumidifierState
[04/06/2022, 12:16:21] [midea-air] Triggered GET TargetHumidifierDehumidifierState
[04/06/2022, 12:16:21] [midea-air] Triggered GET CurrentRelativeHumidity
[04/06/2022, 12:16:21] [midea-air] Triggered GET RelativeHumidityDehumidifierThreshold
[04/06/2022, 12:16:21] [midea-air] Triggered GET WindSpeed
[04/06/2022, 12:16:21] [midea-air] Triggered GET SwingMode
[04/06/2022, 12:16:21] [midea-air] Triggered GET WaterLevel
[04/06/2022, 12:16:21] [midea-air] Triggered GET Active
[04/06/2022, 12:16:21] [midea-air] Triggered GET Current HeaterCooler State
[04/06/2022, 12:16:21] [midea-air] Triggered GET Target HeaterCooler State
[04/06/2022, 12:16:21] [midea-air] Triggered GET CurrentTemperature
[04/06/2022, 12:16:21] [midea-air] Triggered GET ThresholdTemperature
[04/06/2022, 12:16:21] [midea-air] Triggered GET ThresholdTemperature
[04/06/2022, 12:16:21] [midea-air] Triggered GET RotationSpeed
[04/06/2022, 12:16:21] [midea-air] Triggered GET SwingMode
[04/06/2022, 12:16:21] [midea-air] Triggered GET Temperature Display Units
[04/06/2022, 12:16:21] [midea-air] Triggered GET Active
[04/06/2022, 12:16:21] [midea-air] Triggered GET Current HeaterCooler State
[04/06/2022, 12:16:21] [midea-air] Triggered GET Target HeaterCooler State
[04/06/2022, 12:16:21] [midea-air] Triggered GET CurrentTemperature
[04/06/2022, 12:16:21] [midea-air] Triggered GET ThresholdTemperature
[04/06/2022, 12:16:21] [midea-air] Triggered GET ThresholdTemperature
[04/06/2022, 12:16:21] [midea-air] Triggered GET RotationSpeed
[04/06/2022, 12:16:21] [midea-air] Triggered GET SwingMode
[04/06/2022, 12:16:21] [midea-air] Triggered GET Temperature Display Units
[04/06/2022, 12:16:21] [midea-air] Triggered GET Active
[04/06/2022, 12:16:21] [midea-air] Triggered GET Current HeaterCooler State
[04/06/2022, 12:16:21] [midea-air] Triggered GET Target HeaterCooler State
[04/06/2022, 12:16:21] [midea-air] Triggered GET CurrentTemperature
[04/06/2022, 12:16:21] [midea-air] Triggered GET ThresholdTemperature
[04/06/2022, 12:16:21] [midea-air] Triggered GET ThresholdTemperature
[04/06/2022, 12:16:21] [midea-air] Triggered GET RotationSpeed
[04/06/2022, 12:16:21] [midea-air] Triggered GET SwingMode
[04/06/2022, 12:16:21] [midea-air] Triggered GET Temperature Display Units
[04/06/2022, 12:16:21] [midea-air] Send command to: Bedroom AC (43980465157819) [sendUpdateToDevice] attempt 1/2 success!
[04/06/2022, 12:16:21] [midea-air] useFahrenheit is set to: false
[04/06/2022, 12:16:21] [midea-air] Target Temperature: 24˚C
[04/06/2022, 12:16:21] [midea-air] Indoor Temperature is: 20.5˚C
[04/06/2022, 12:16:21] [midea-air] Outdoor Temperature is: 20˚C
[04/06/2022, 12:16:21] [midea-air] Power State is set to: 0
[04/06/2022, 12:16:21] [midea-air] Operational Mode is set to: 1
[04/06/2022, 12:16:21] [midea-air] Fan Speed is set to: 102
[04/06/2022, 12:16:21] [midea-air] Swing Mode is set to: 12
[04/06/2022, 12:16:21] [midea-air] Eco Mode is set to: false
[04/06/2022, 12:16:21] [midea-air] Turbo Mode is set to: false
[04/06/2022, 12:16:21] [midea-air] Full data is: 11000000,0,101000,1100110,0,0,0,111100,0,0,0,1011011,1011010,0,1110000,0,0,0,0,0,0,0,0,0,11101000,10010101,0,0,0,0,0,0,0,0,0
[04/06/2022, 12:16:21] [midea-air] [sendUpdateToDevice] Send command to device: Bedroom AC (43980465157819)`
```

**Comment by hillaliy** [comment]
The commands on the new Midea devices (U, Balnc etc....) are a little different.
I can fix it only if I have the Midea API protocol.
About scrolling past every 30 seconds it's supposed to be like this.
The plugin ask the device every 30 seconds of it status. You can change it on advance settings but the state will be less accurate

**Comment by hillaliy** [comment]
I release new stable version of fixing the Led screen. 1.5.2

**Comment by noltondyer** [comment]
How would you be able to add support for the U. Is there any way I could help out?

**Comment by hillaliy** [comment]
If you can get the Midea Protocol. The old protocol was leaked to the net.

**Comment by cdkii** [comment]
I’m guessing there’s nothing useful on the Midea IoT Developer Platform (https://mis.midea.com/#/pre-login) since that seems to be more more ‘official’ access to their cloud platform. Also doesn’t help having to auto-translate everything from Chinese to figure out what’s going on on that site.

**Comment by hillaliy** [comment]
most of the Midea U devices don't work with this plugin. I think it's the new dongles OSK105.
What's your dongle model?

**Comment by cdkii** [comment]
Yep. Just checked and all three are US-SK105.  (Weirdly two are on firmware v. 1.0.7 and the newest one is on v. 150023082122. I checked for updates and all said they're on the newest firmware. So no idea on that.)  

Everything works perfectly in HomeKit on these three Midea U units minus the unit getting toggled to Celsius on the LED display and in the Midea Air app. It continues to show in the Apple Home app as Fahrenheit, so my current workaround is just to leave the LED display off on the units. 

I honestly appreciate all your work to get these working with HomeKit, even if it isn't fully supported. I read an article earlier this year that Midea plans to support the Matter standard, so I'm hoping they upgrade the existing models and that that will bring native compatibility with HomeKit. I wouldn't be surprised if it's only new models though.

**Comment by hillaliy** [comment]
Fix on 1.5.5
